src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -86,6 +86,10 @@
     ("exclude", "StartExcludeBracketSegment", "EndExcludeBracketSegment", True)
 )
 
+# In Snowflake, FLATTEN() returns a "value table".
+# https://docs.snowflake.com/en/user-guide/json-basics-tutorial-flatten.html
+snowflake_dialect.sets("value_table_functions").update(["flatten"])
```

Comment:
Because this was missing, [this code](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/core/rules/analysis/select.py#L137L138) was not executing, preventing the `FLATTEN()` call from being recognized as returning a table.